### PR TITLE
Fix LINQ / List alllocations in PropagateBlockableEvent

### PR DESCRIPTION
**Pull request by peppy**
Two small changes:
- Don't alloc lists for `IRequireHighFrequency`
- Don't alloc enumerator in `PropagateBlockingEvent` (due to `IEnumerable` param type)
- [x] Depends on https://github.com/ppy/osu-framework/pull/3739.

As one requires the other I did this in a single PR.

Before:
image

After:
image

### CODE REVIEW

File: osu.Framework/Input/InputManager.cs [hunk 1]
```diff
@@ -385,7 +387,17 @@ protected override void Update()
 
             if (CurrentState.Mouse.IsPositionValid)
             {
-                PropagateBlockableEvent(PositionalInputQueue.Where(d => d is IRequireHighFrequencyMousePosition), new MouseMoveEvent(CurrentState));
+                highFrequencyDrawables.Clear();
```
Comment (bdach) [inline osu.Framework/Input/InputManager.cs:390]: The double clear is a little bit weird to me. Is there a particular reason for this? I'd probably either just clear once at the start of the method (although that could hold references for a frame) or clear at the end and make this an assert.

Comment (peppy) [reply osu.Framework/Input/InputManager.cs:390]: Assert sounds good. I added this here to drop references immediately, so would probably go with that direction (assert before, clear after).

File: osu.Framework/Input/InputManager.cs [hunk 2]
```diff
@@ -687,7 +699,7 @@ protected virtual void HandleMouseButtonStateChange(ButtonStateChangeEvent<Mouse
         /// <param name="drawables">The drawables in the queue.</param>
         /// <param name="e">The event.</param>
         /// <returns>Whether the event was handled.</returns>
-        protected virtual bool PropagateBlockableEvent(IEnumerable<Drawable> drawables, UIEvent e)
+        protected virtual bool PropagateBlockableEvent(List<Drawable> drawables, UIEvent e)
```
Comment (smoogipoo) [inline osu.Framework/Input/InputManager.cs:702]: See CI failure. `IList<Drawable>` is probably good here to keep the interface simple (and iterate via index).

Comment (peppy) [reply osu.Framework/Input/InputManager.cs]: I've made is `Slim`, not `IList` because I don't think there will be any other usages.

**Pull request merged**